File: code-examples/pymongo/usage-examples/deleteMany.py
# Delete multiple documents

import pymongo

# Replace the uri string with your MongoDB deployment's connection string
uri = "<connection string uri>"

client = pymongo.MongoClient(uri)


def run():
    try:
        # Access the 'sample_mflix' database and 'movies' collection
        database = client['sample_mflix']
        movies = database['movies']

        # Delete all documents that match the specified regular
        # expression in the title field from the 'movies' collection
        result = movies.delete_many({'title': {'$regex': 'Santa'}})

        # Print the number of deleted documents
        print(f'Deleted {result.deleted_count} documents')
    except Exception as e:
        print(e)
    finally:
        # Close the connection after the operation completes
        client.close()


# Run the program
run()

File: code-examples/php/usage-examples/deleteMany.php
<?php
// Delete multiple documents
require 'vendor/autoload.php'; // Include Composer's autoloader

// Replace the uri string with your MongoDB deployment's connection string
$uri = "mongodb://localhost:27017";

$client = new MongoDB\Client($uri);

try {
    $collection = $client->sample_mflix->movies;

    // Delete all documents that match the specified regular expression in the title field from the "movies" collection
    $regexFilter = new MongoDB\BSON\Regex('Santa', 'i');
    $result = $collection->deleteMany([ 'title' => $regexFilter ]);

    // Print the number of deleted documents
    echo "Deleted " . $result->getDeletedCount() . " documents\n";
} catch (Exception $e) {
    // Print any thrown exceptions
    echo $e->getMessage(), "\n";
}

?>
